### PR TITLE
Release Notes + Highlights for 2.11.1

### PR DESCRIPTION
**Pull request by naemono**
This adds release notes and highlights for the upcoming 2.11.1 release.
Also added known issues section in 2.11.0 highlights.

### CODE REVIEW

File: docs/release-notes/highlights-2.11.0.asciidoc [hunk 1]
```diff
@@ -1,6 +1,11 @@
 [[release-highlights-2.11.0]]
 == 2.11.0 release highlights
 
+[float]
+[id="{p}-2110-known-issues"]
+=== Known issues
+- The `resourceStatuses` field of the status subresource of the Stack Configuration Policy was removed which could cause errors in the operator logs during an upgrade.
```
Comment (naemono) [inline docs/release-notes/highlights-2.11.0.asciidoc:7]: I'm unsure about this wording, and how we would prefer to word this. Comments appreciated.

**Comment by naemono** [comment]
CI is noting changes such as these are needed, but I can't seem to get this format to generate locally. Any ideas?
```
-                      description: CurrentVersion denotes the current version of filesettings
-                        applied to the Elasticsearch cluster This field does not apply
-                        to Kibana resources
+                      description: \|-
+                        CurrentVersion denotes the current version of filesettings applied to the Elasticsearch cluster
+                        This field does not apply to Kibana resources
```

**Review by barkbay — COMMENTED** [review]
Maybe update the statement [here](https://github.com/elastic/cloud-on-k8s/blob/76c6265ebef1024824880e03891630f5e1d06b54/docs/release-notes/2.11.0.asciidoc?plain=1#L11):

> [...] there should be no impact on users [...]

To mention what should be a the temporary disruption when installing `2.11.0` and `2.11.1` has been released to workaround that?

**Comment by barkbay** [comment]
> CI is noting changes such as these are needed, but I can't seem to get this format to generate locally. Any ideas?

Yes, this: https://github.com/kubernetes-sigs/controller-tools/pull/870, which was introduced in [the recently updated](https://github.com/elastic/cloud-on-k8s/pull/7461) `v0.14.0`. It leads to the following files being updated on `main` if you run `make generate`: 

```
On branch main
Your branch is up to date with 'origin/main'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
        modified:   config/crds/v1/all-crds.yaml
        modified:   config/crds/v1/bases/stackconfigpolicy.k8s.elastic.co_stackconfigpolicies.yaml
        modified:   deploy/eck-operator/charts/eck-operator-crds/templates/all-crds.yaml
```

File: docs/release-notes/highlights-2.11.0.asciidoc (hunk 1 above)
Comment (pebrc) [inline docs/release-notes/highlights-2.11.0.asciidoc:7]:
```suggestion
- The `resourceStatuses` field of the status subresource of the Stack Configuration Policy was removed which could cause errors when running older versions of the operator with the new CRDs during upgrades.
```

**Review by pebrc — COMMENTED** [review]
+ 1 on updating the breaking changes statement for 2.11.0 as well.

File: docs/release-notes/2.11.0.asciidoc [hunk 2]
```diff
@@ -8,7 +8,7 @@
 [float]
 === Breaking changes
 
-* The `resourceStatuses` field of the status subresource of the Stack Configuration Policy is no longer in use. Instead a new `details` field is populated which now also contains information about configured Kibana applications. Unless users have built automation that parses the status subresource, there should be no impact on users, as these fields serve only informational purposes. {pull}7433[#7433]
+* The `resourceStatuses` field of the status subresource of the Stack Configuration Policy is no longer in use. Instead a new `details` field is populated which now also contains information about configured Kibana applications. This change could cause errors during an upgrade of the CRDs while your operator is still running an older version but 2.11.1 was released to mitigate this issue. {pull}7433[#7433]
```
Comment (barkbay) [inline docs/release-notes/2.11.0.asciidoc:11]: nit:

```suggestion
* The `resourceStatuses` field of the status subresource of the Stack Configuration Policy is no longer in use. Instead a new `details` field is populated which now also contains information about configured Kibana applications. This change could cause errors during an upgrade of the CRDs while your operator is still running an older version, 2.11.1 has been released to mitigate this issue. {pull}7433[#7433]
```

(not a native speaker, feel free to ignore 😄 )

**Review by barkbay — APPROVED** [review]
LGTM

File: docs/release-notes/2.11.0.asciidoc (hunk 2 above)
Comment (naemono) [reply docs/release-notes/2.11.0.asciidoc]: > (not a native speaker, feel free to ignore 😄 )

Nope, I see your point here, but I added a `however` after the comma, as it made it flow a bit better.

**Comment by naemono** [comment]
@elasticmachine run elasticsearch-ci/doc

**Comment by naemono** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|2.11.1-wip|[](https://github.com/elastic/cloud-on-k8s/pull/7513)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Pull request merged**